IPython/html/widgets/widget_float.py

```diff
@@ -38,14 +38,32 @@ class _BoundedFloat(_Float):
     def __init__(self, *pargs, **kwargs):
         """Constructor"""
         super(_BoundedFloat, self).__init__(*pargs, **kwargs)
-        self._validate('value', None, self.value)
-        self.on_trait_change(self._validate, ['value', 'min', 'max'])
+        self._validate_value('value', None, self.value)
+        self._handle_min_changed('min', None, self.value)
+        self._handle_max_changed('max', None, self.value)
+        self.on_trait_change(self._validate_value, ['value'])
+        self.on_trait_change(self._handle_max_changed, ['max'])
+        self.on_trait_change(self._handle_min_changed, ['min'])
 
-    def _validate(self, name, old, new):
-        """Validate value, max, min."""
+    def _validate_value(self, name, old, new):
+        """Validate value."""
         if self.min > new or new > self.max:
             self.value = min(max(new, self.min), self.max)
 
+    def _handle_max_changed(self, name, old, new):
```

Comment:
Why not use the `_max_changed` name instead of defining the method and registering it manually?